config/jobs/kubernetes/sig-autoscaling/azure/cluster-autoscaler-azure-presubmits.yaml

```diff
@@ -25,25 +27,39 @@ presubmits:
     spec:
       serviceAccountName: azure
       containers:
-      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20240803-cf1183f2db-1.29
-        # This job is being phased out in favor of pull-cluster-autoscaler-e2e-azure-master. It always
-        # passes in order not to block PRs where the previous version of this job failed. This job can
-        # be removed when it is passing on all open PRs.
```

Comment:
This PR now removes this old job and adds the 1.31 branch job. I split this into two commits to make that a little easier to follow.